api/src/main/java/com/cloud/resource/RollingMaintenanceManager.java

```diff
@@ -0,0 +1,146 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package com.cloud.resource;
+
+import com.cloud.host.Host;
+import com.cloud.utils.Pair;
+import com.cloud.utils.Ternary;
+import com.cloud.utils.exception.CloudRuntimeException;
+import org.apache.cloudstack.api.command.admin.resource.StartRollingMaintenanceCmd;
+import org.apache.cloudstack.framework.config.ConfigKey;
+import org.apache.cloudstack.framework.config.Configurable;
+
+import java.util.Date;
+import java.util.List;
+
+public interface RollingMaintenanceManager extends Configurable {
+
+    ConfigKey<Integer> KvmRollingMaintenanceStageTimeout = new ConfigKey<>("Advanced", Integer.class,
+            "kvm.rolling.maintenance.stage.timeout", "1800",
+            "Wait timeout (in seconds) for a rolling maintenance stage update from hosts",
+            true, ConfigKey.Scope.Global);
+    ConfigKey<Integer> KvmRollingMaintenancePingInterval = new ConfigKey<>("Advanced", Integer.class,
+            "kvm.rolling.maintenance.ping.interval", "10",
+            "Ping interval in seconds between management server and hosts performing stages during rolling maintenance",
+            true, ConfigKey.Scope.Global);
+    ConfigKey<Integer> KvmRollingMaintenanceWaitForMaintenanceTimeout = new ConfigKey<>("Advanced", Integer.class,
+            "kvm.rolling.maintenance.wait.maintenance.timeout", "600",
```

Comment:
1800 - for beefy hosts with lot's of VMs etc - it makes take some time to migrate VMs away - and the "Maintenance" state will not be reached maybe within 10min/600sec

Reply:
Done, thanks